### PR TITLE
Fix mtar file size truncation

### PR DESCRIPTION
**Pull request by oxidase**
# Issue

Fixes #5010.

/cc @TheMarex 

Please read our [documentation](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/releasing.md) on release and version management.
If your PR is still work in progress please attach the relevant label.

## Tasklist

 - [x] test against large OSM pbf
 - [x] add tests (see [testing documentation](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/testing.md)
 - [x] review
 - [x] adjust for comments
 - [ ] cherry pick to release branch `5.17`

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

**Comment by oxidase** [comment]
@TheMarex I've just dumped `.osrm` file
```
00000400  2f 65 78 74 72 61 63 74  6f 72 2f 6e 6f 64 65 73  |/extractor/nodes|
00000410  2e 6d 65 74 61 00 00 00  00 00 00 00 00 00 00 00  |.meta...........|
00000420  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
*
00000460  00 00 00 00 36 36 34 00  00 00 00 00 30 00 00 00  |....664.....0...|
00000470  00 00 00 00 00 00 00 00  00 00 00 00 31 30 00 00  |............10..|
00000480  00 00 00 00 00 00 00 00  30 00 00 00 00 00 00 00  |........0.......|
00000490  00 00 00 00 30 30 35 32  37 31 00 20 30 00 00 00  |....005271. 0...|
000004a0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
*
00000600  bd fa d4 1c 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000610  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
*
00000800  2f 65 78 74 72 61 63 74  6f 72 2f 6e 6f 64 65 73  |/extractor/nodes|
00000810  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
*
00000860  00 00 00 00 36 36 34 00  00 00 00 00 30 00 00 00  |....664.....0...|
00000870  00 00 00 00 00 00 00 00  00 00 00 00 33 31 35 32  |............3152|
00000880  33 37 32 35 37 32 30 00  30 00 00 00 00 00 00 00  |3725720.0.......|
00000890  00 00 00 00 30 30 35 32  37 30 00 20 30 00 00 00  |....005270. 0...|
000008a0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
```

and checked that the nodes blocks size  `/extractor/nodes.meta` 0x1cd4fabd = 483719869 * sizeof(QueryNode) does not correspond to the block size in the `/extractor/nodes` header that equals  `(483719869 * 16) & 0xffffffff]= 3444550608` or 31523725720 in octal.

**Comment by TheMarex** [comment]
Locally I'm seeing the following failure when preparing the data.

```
[error] I/O error occurred: mld/monaco.osrm.cnbg :  (at include/storage/tar.hpp:33)
[error] I/O error occurred: monaco.osrm :  (at include/storage/tar.hpp:33)
```

File: third_party/microtar/src/microtar.c [hunk 1]
```diff
@@ -105,24 +106,68 @@ static int raw_to_header(mtar_header_t *h, const mtar_raw_header_t *rh) {
...
 
 static int header_to_raw(mtar_raw_header_t *rh, const mtar_header_t *h) {
   unsigned chksum;
+  mtar_size_t filesize = h->size;
 
   /* Load header into raw header */
   memset(rh, 0, sizeof(*rh));
+
+  /* Store size in ASCII octal digits or base-256 formats */
+  if (sizeof(mtar_size_t) <= 4 || filesize <= (mtar_size_t)0777777777777LL) {
```
Comment (TheMarex) [inline third_party/microtar/src/microtar.c]: I'm counting 12 x `7` here, but for fitting into 12 bytes we can only save 11 x `7` plus one `\0` character?

Comment (oxidase) [reply third_party/microtar/src/microtar.c]: It is intentional and probably is a GNU tar extension of POSIX tar format.
GNU tar uses all 12 characters of `header.size` in [`OFF_FROM_HEADER`](https://git.savannah.gnu.org/cgit/tar.git/tree/src/common.h#n605) that is the `digs` argument  of [`from_header`](https://git.savannah.gnu.org/cgit/tar.git/tree/src/list.c#n727) function.

Anyway it is possible to drop one 7  and have only proper ASCIIZ strings.

**Pull request merged**